web/src/App.tsx

```diff
@@ -1,32 +1,46 @@
-import React from 'react'
-import { BrowserRouter, Routes, Route } from 'react-router-dom'
+import React from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
 
-import GamesList from './views/GamesList'
-import Game from './views/Game'
-import GameEdit from './views/Game/Edit'
+import GamesList from "./views/GamesList";
+import Game from "./views/Game";
+import GameEdit from "./views/Game/Edit";
 
-import StatsPage from './views/StatsPage'
-import Leaderboards from './views/Leaderboards'
-import ComparePlayers from './views/ComparePlayers'
+import StatsPage from "./views/StatsPage";
+import Leaderboards from "./views/Leaderboards";
+import ComparePlayers from "./views/ComparePlayers";
 
-import SalaryPage from './views/SalaryPage'
-import StatsTable from './views/StatsTable'
-import TeamDashboard from './views/TeamDashboard'
+import SalaryPage from "./views/SalaryPage";
+import StatsTable from "./views/StatsTable";
+import TeamDashboard from "./views/TeamDashboard";
+import { LeaderboardsContextProvider } from "./views/Leaderboards/LeaderboardsContext";
 
 function App() {
   return (
     <BrowserRouter>
       <Routes>
         <Route path="/" element={<StatsPage component={StatsTable} />} />
-        <Route path="/games" element={<GamesList/>} />
-        <Route path="/:leagueId/games/:gameId" element={<Game/>} />
-        <Route path="/:leagueId/games/:gameId/edit" element={<GameEdit/>} />
-        <Route path="/leaderboards" element={<StatsPage component={Leaderboards} />} />
-        <Route path="/compare_players" element={<StatsPage component={ComparePlayers}/>} />
-        <Route path="/team_dashboard" element={<SalaryPage component={TeamDashboard} />} />
+        <Route path="/games" element={<GamesList />} />
+        <Route path="/:leagueId/games/:gameId" element={<Game />} />
+        <Route path="/:leagueId/games/:gameId/edit" element={<GameEdit />} />
+        <Route
+          path="/leaderboards"
+          element={
+            <LeaderboardsContextProvider>
```

Comment:
I am not sure this extra component is helping all that much both here and inside the leaderboard component. I think it would keep the app simpler for less react inclined folk to add this logic right to the leaderboard component. If you really think it helps it at least needs to be re-worked so we are not adding logic in this main file.

Also did you consider adding this to the StatsPage component? I seem to recall its purpose is pretty much the same - to fetch and then filter the stats.